### ISSUE TITLE
Validation step super slow with ddp

### ISSUE THREAD

**Issue by mnslarcher** [Lightning-AI/lightning#15142]
### Bug description

I'm using Pytorch Lightning with SageMaker on a ml.p3.8xlarge. I experienced an important increase in validation time so I re-executed an old experiment and still I continue to see a large increase in val time, something seems wrong:

I use:
```
"trainer.strategy": "ddp"
"trainer.sync_batchnorm": "true"
```

Before this was a train log at Epoch 0, 75% (1.24it/s):
Epoch 0:  75%|███████▍  | 430/577 [05:47<01:58,  1.24it/s, loss=0.443, v_num=0]

and this a val log at 20% (3.39it/s):
Validation DataLoader 0:  20%|██        | 13/65 [00:15<00:15,  3.39it/s]

Now I get this (I launched the same job for test):

Train log at Epoch 0, 75% (1.28it/s): <- more or less the same
Epoch 0:  75%|███████▍  | 430/577 [05:35<01:54,  1.28it/s, loss=0.452, v_num=0]

val log at 20% (68.74s/it): <- 20x slower
Validation DataLoader 0:  20%\|██        \| 13/65 [14:53<59:34, 68.74s/it]

As you can see now val is extremely slow, both compared to train and to what was the speed before.

What I'm missing?

I can do other experiments but it would be super useful to have some guidance on why this can happen.

FIrst exp was with pytorch_lightning-1.6.4
Second (new) exp is with pytorch_lightning-1.7.7

Best,
M

**Comment by awaelchli** [comment]
@mnslarcher Is the only thing that changed from first to second experiment the version Lightning? There could be many reasons why validation is slow. I would start disabling everything that's not strictly needed, like logging in val step for example.

**Comment by mnslarcher** [comment]
> @mnslarcher Is the only thing that changed from first to second experiment the version Lightning? There could be many reasons why validation is slow. I would start disabling everything that's not strictly needed, like logging in val step for example.

Hi @awaelchli , I experimented some more, this time copying code that had been used in a successful job (SageMaker zippers the code when launching a job with the train) I also fixed the libraries that include pytorch-lightning and still the problem persists.

At this point my best guess is that something has changed in the Pytorch container I'm using (763104351884.dkr.ecr.eu-central-1.amazonaws.com/pytorch-training:1.9.1-gpu-py38-cu111-ubuntu20.04) that produces this strange behavior whereby the train runs fine but the validation takes forever.

I don't know if it is related, but another problem we have experienced in the past is that trainer.test() blocks the job.

If I find the problem I will update this topic in case it might be useful to others.

**Comment by awaelchli** [comment]
@mnslarcher I thought about it but I don't have any good guesses on what might be the problem. I ran a quick test with our debugging model but couldn't find anything unusual. I couldn't use your image though, because it requires credentials. Would you be able to share a reproducible code. 

One reason validation runs slow(er) could be because of collective calls (all_gather, reduce, etc.) that happen in validation_step(). Are you using torchmetrics, or self.log?

**Comment by mnslarcher** [comment]
@awaelchli : we tried a lot of things but nothing, and now... it is OK again without changing a line of code. I think it is related to what AWS is running in it's containers or I have no idea what it could be, we also tried removing all logs and metrics from val but the problem remained (now it is gone again using the old code)

**Comment by awaelchli** [comment]
We can reopen again if this comes up again and needs investigation.
